File: government/src/main/java/service/VehicleService.java
package service;

import dao.VehicleDao;
import domain.Location;
import domain.Vehicle;
import domain.enums.VehicleType;

import javax.ejb.Stateless;
import javax.inject.Inject;
import java.util.List;

@Stateless
public class VehicleService {

    @Inject
    VehicleDao vehicleDao;

    public List<Vehicle> getAllVehicles() {
        return vehicleDao.getAllVehicles();
    }

    public List<Vehicle> getAllVehiclesInCountry(String countryName) {
        return vehicleDao.getAllVehiclesInCountry(countryName);
    }

    public Vehicle addVehicle(String hardwareSerialNumber, VehicleType vehicleType,
                              String licensePlate) {
        return vehicleDao.addVehicle(new Vehicle(hardwareSerialNumber, vehicleType,
                new Location(), licensePlate));
    }
}

File: government/src/main/java/service/VehicleService.kt
package service

import dao.VehicleDao
import domain.Location
import domain.Vehicle
import domain.enums.VehicleType

import javax.ejb.Stateless
import javax.inject.Inject

@Stateless
class VehicleService @Inject constructor(
        val vehicleDao: VehicleDao
){
    val allVehicles: List<Vehicle>
        get() = vehicleDao.allVehicles

    fun getAllVehiclesInCountry(countryName: String): List<Vehicle> {
        return vehicleDao.getAllVehiclesInCountry(countryName)
    }

    fun addVehicle(hardwareSerialNumber: String, vehicleType: VehicleType, licensePlate: String): Vehicle {
        return vehicleDao!!.addVehicle(Vehicle(hardwareSerialNumber, vehicleType, Location(), licensePlate))
    }
}
